File: src/PostAggregator.Api/Data/Entities/Post.cs
namespace PostAggregator.Api.Data.Entities;

public class Post
{
    public Guid Id { get; set; }
    public string Title { get; set; } = string.Empty;
    public string Author { get; set; } = string.Empty;
    public DateTime CreatedAtUtc { get; set; }
    public string? Link { get; set; }
    public string? Thumbnail { get; set; }
    public Source Source { get; set; }
    public string? Text { get; set; }
}

File: src/PostAggregator.App/src/types/api/Post.ts
export interface Post {
  id: string; // Guid is typically a string in API responses
  title: string;
  author: string;
  createdAtUtc: string; // This can be a string (ISO 8601 date format) or a Date object, depending on how it's returned
  link: string;
  thumbnail: string;
  text?: string;
}
